example-workflows/.github/workflows/cloud-functions.yml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build and Deploy to Cloud Functions
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  # GCP_PROJECT is defined in service_account_key
+  PROJECT_ID: ${{ secrets.GCP_PROJECT }}
+  FUNCTION_REGION: northamerica-northeast1
+
+jobs:
+  setup-build-deploy:
+    name: Setup, Build, and Deploy
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    # Setup gcloud CLI
+    - uses: google-github-actions/setup-gcloud@master
+      with:
+        service_account_key: ${{ secrets.GCP_SA_KEY }}
+        export_default_credentials: true
+    - id: Deploy
+      uses: google-github-actions/deploy-cloud-functions@main
```

Comment:
As this is a native action, we should prefer authenticating natively via `credentials` if we are not using gcloud in any subsequent steps.